### ISSUE TITLE
[Question/Issue] Unable to use a single file to route index.tsx `/` and `/*` simultaneously

### ISSUE THREAD

**Issue by mcandeia** [denoland/fresh#1124]
# Description:

I'm using Fresh and I want to create a route that matches URLs with only a slash at the end (/) and URLs that contain any character after the slash (/any-character). According to the Fresh documentation, this can be done by creating two files (index.tsx and [...catchall].tsx) and using a catch-all route.

However, I don't want to duplicate code in the index.tsx and [...catchall].tsx files, since they should contain the same content (also don't want to import/export again, because I'm trying to generate it automatically (creating my own `dev.ts`)). Is there a way to use a single file for both routes ?

Additionally, I've noticed that using routeOverride to override a catch-all route can break the sort algorithm in Fresh, preventing the "static" routes from being properly sorted. Is there a way to use routeOverride without breaking the sort algorithm?

What I've tried:

```ts
export const config = {
  routeOverride: "/*",
};
```

It worked... but the sort algorithm mark this route as a higher priority (actual equal, but now it depends on the order defined at `routes` in the fresh.gen.ts file) than `/a/b/c` for instance.

Thanks in advance for any help or suggestions on how to make this work properly.

**Comment by lucacasonato** [comment]
To fix the sort ordering, rename `/index.tsx` to `/[...catchall].tsx`. This will cause it to get the lowest possible priority during sorting.

The `routeOverride` looks correct.

**Comment by deer** [comment]
Hey @lucacasonato and @mcandeia, I wrote https://fresh.deno.dev/docs/examples/handling-complex-routes earlier this week. I think this example here would be a fantastic addition to the page, but I sadly don't understand, and therefore can't reproduce.

My `routes` folder looks like this:
```
.
├── [...catchall].tsx
└── a
    └── b
        └── index.tsx
```
and `[...catchall].tsx` has this config:
```ts
export const config = {
  routeOverride: "/*",
};
```

When I go to `http://localhost:8000/a/b` I'm routed to the catchall, unless I modify the `fresh.gen.ts` file to have a different routing order. Have I reproduced the functional case correctly, but I'm doing something wrong with the routes? Or am I misunderstanding the use case?

Edit: when I modify `sortRoutes` to log what it's doing (i.e. each comparison and the value), I get the following for my simple case:
```
/a/b /* 0
```
So I don't understand how renaming does anything, since the file has a `routeOverride` and then the name is irrelevant.
